design/cluster-scope-resource-filter.md

```diff
@@ -0,0 +1,572 @@
+
+* `--exclude-namespaces` also supports `*`.
+
+* `--exclude-cluster-scope-resources` valid value is comma separated string.
+
+* `--include-cluster-scope-resources` and `--exclude-cluster-scope-resources` can only contain cluster scope resource type names. If namespace scope resource type names are included, validation failure should be returned.
+
+* If there are conflicts between `--include-cluster-scope-resources` and `--exclude-cluster-scope-resources` specified resources type lists, validation failure should be returned.
+
+* `--include-namespaced-resources` and `--exclude-namespaced-resources` can only contain namespace scope resource type names. If cluster scope resource type names are included, validation failure should be returned.
+
+* If there are conflicts between `--include-namespaced-resources` and `--exclude-namespaced-resources` specified resources type lists, validation failure should be returned.
+
+* If  `--include-namespaced-resources` is not present, it means including all namespace scope resources per resource type.
+
+* If `--include-cluster-scope-resources` is not present, it means no additional cluster resource is included per resource type.
+
+* `--include-cluster-scope-resources`, `--exclude-cluster-scope-resources`, `--include-namespaced-resources` and `--exclude-namespaced-resources` have interact with `--include-resources` and `--exclude-resources`. The rule is specified in [IncludedResources and ExcludedResources change](#includedresources-and-excludedresources-change)
+
+* `--include-cluster-scope-resources` and `--exclude-cluster-scope-resources` have interact with `--include-cluster-resources` parameter. The rule is specified in [IncludeClusterResources change](#includeclusterresources-change)
+
+* If the backup or restore contains no resource, validation failure should be returned.
+
+### Using scenarios:
+#### no namespaced resources + no cluster resources
+This is not allowed. Backup or restore cannot contain no resource.
+
+#### no namespaced resources + some cluster resources
+The following commands all mean backup no namespaced resources and some cluster resources.
+
+``` bash
+velero backup create <backup-name>
+	--exclude-namespaces=*
+	--include-cluster-resources=true
+	--include-cluster-scope-resources=StorageClass
+```
+
+``` bash
+velero backup create <backup-name>
+	--exclude-namespaced-resources=*
+	--include-cluster-resources=true
+	--include-cluster-scope-resources=StorageClass
+```
+
+``` bash
+velero backup create <backup-name>
+	--exclude-namespaces=*
+	--exclude-namespaced-resources=*
+	--include-cluster-resources=true
+	--include-cluster-scope-resources=StorageClass
+```
+
+#### no namespaced resources + all cluster resources
+The following commands all mean backup no namespaced resources and all cluster resources.
+
+``` bash
+velero backup create <backup-name>
+	--exclude-namespaces=*
```

Comment:
I don't think we need this -- just make sure `--include-namespace-resources` is absent/empty.

Reply:
Also note, this will include no cluster-scoped resources unless `--include-cluster-scope-resources=*` is added.